qiskit/_instructionset.py

```diff
@@ -17,19 +17,19 @@ class InstructionSet(object):
 
     def __init__(self):
         """New collection of instructions."""
-        self.instructions = set([])
+        self.instructions = []
```

Comment:
To be able to replace an instruction in the instruction set. See the next hunk:
```
-        for instruction in self.instructions:
-           instruction.inverse()
+        for index, instruction in enumerate(self.instructions):
+           self.instructions[index] = instruction.inverse()
```
Because now the inverse of S is of a different type, you need to replace the instruction (it cannot be changed "on the fly", like before). Replace elements in a set is hacky. The `instractionset` was never a set and how the things a store internally does not matter much because we do not expose (correctly) that internal structure.